src/access/workflows/WithdrawWorkflow.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IAccessPoint} from "../../interfaces/IAccessPoint.sol";
+
+interface IWrappedNativeAsset is IERC20 {
+    function deposit() external payable;
+
+    function withdraw(uint256 amount) external;
+}
+
+contract WithdrawWorkflow {
+    using SafeERC20 for IERC20;
+
+    error NativeWithdrawalFailed();
+
+    function withdrawERC20(IERC20 asset, uint256 amount) external {
+        address owner = _getOwner();
+        asset.safeTransfer({to: owner, value: amount});
+    }
+
+    function withdrawNative(uint256 amount) external {
+        address owner = _getOwner();
+        (bool sent,) = owner.call{value: amount}("");
+        if (!sent) {
+            revert NativeWithdrawalFailed();
+        }
+    }
+
+    function _getOwner() internal view returns (address) {
```

Comment:
nit pick: maybe we can just not have this method and make withdrawNative and withdrawERC20 access directly `IAccessPoint(address(this)).owner()` without having to call the method (not strongly opinionated though)

Reply:
I believe it is quite irrelevant, so I have no objection to making changes. However, given its insignificance, I won't be doing it myself.